### PR TITLE
Update centos dockerfiles, for openshift images.

### PR DESCRIPTION
**Pull request by mohammedzee1000**
Update dockerfiles to build on top of centos container pipeline from
registry.centos.org

Updated custom docker builder to use 1.5 as base.

Updated HA Proxy container to use 1.5 as base.

Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
ping @tdawson @stevekuznetsov

**Comment by tdawson** [comment]
👍 
Looks great.

File: images/base/Dockerfile.centos7 [hunk 1]
```diff
@@ -4,9 +4,9 @@
 #
```
Comment (stevekuznetsov) [inline images/base/Dockerfile.centos7]: Why does `images/base/Dockerfile` not work for you? What is different about this `.centos7` version that you need?

File: images/builder/docker/custom-docker-builder/Dockerfile [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #
 # The standard name for this image is openshift/origin-custom-docker-builder
 #
-FROM openshift/origin-base
+FROM openshift/origin-base:v1.5
```
Comment (stevekuznetsov) [inline images/builder/docker/custom-docker-builder/Dockerfile]: Do you have the ability to overwrite `FROM`? Maybe when you push to dist-git? As a rule we don't use versioned `FROM` in these `Dockerfiles`

File: images/base/Dockerfile.centos7 (hunk 1 above)
Comment (mohammedzee1000) [inline images/base/Dockerfile.centos7]: Noted, will update accordingly. I too just noticed little difference.

File: images/builder/docker/custom-docker-builder/Dockerfile (hunk 2 above)
Comment (mohammedzee1000) [inline images/builder/docker/custom-docker-builder/Dockerfile]: I dont think we do atm, @bamachrn you want to comment on this?

File: images/base/Dockerfile.centos7 (hunk 1 above)
Comment (mohammedzee1000) [inline images/base/Dockerfile.centos7]: Ok i believe the only difference is the rpm. @tdawson you ok with merging these two.

File: images/builder/docker/custom-docker-builder/Dockerfile (hunk 2 above)
Comment (bamachrn) [inline images/builder/docker/custom-docker-builder/Dockerfile]: We do not have the ability to alter FROM as of now. But I see the point of putting specific FROM for versioned images. @mohammedzee1000 these dockerfiles are specific to v1.5 right? then how this is getting build now? taking base as latest?

Comment (mohammedzee1000) [inline images/builder/docker/custom-docker-builder/Dockerfile]: Hmm, i believe so. This is where part of problem comes for current images getting updated.

File: images/base/Dockerfile.centos7 (hunk 1 above)
Comment (stevekuznetsov) [inline images/base/Dockerfile.centos7]: At this point I am confident that if `images/base/Dockerfile` works for you, we should merge them and remove the CentOS one (and all of the others)

File: images/builder/docker/custom-docker-builder/Dockerfile (hunk 2 above)
Comment (stevekuznetsov) [inline images/builder/docker/custom-docker-builder/Dockerfile]: As I understand it, `FROM` stanza is updated in dist-git before images are built by OSBS. These files aren't taken verbatim to build. You'll notice they are also missing the OSBS-specific labels

File: images/base/Dockerfile.centos7 (hunk 1 above)
Comment (tdawson) [inline images/base/Dockerfile.centos7]: I think merging them would be great.
I am assuming that "merging them" means have the regular Dockerfile look like Dockerfile.centos7 and not the other way around.

Comment (stevekuznetsov) [inline images/base/Dockerfile.centos7]: Agreed, we should be able to remove `.centos7` and update the normal ones to match.

Comment (mohammedzee1000) [inline images/base/Dockerfile.centos7]: Updated.

File: images/builder/docker/custom-docker-builder/Dockerfile (hunk 2 above)
Comment (mohammedzee1000) [inline images/builder/docker/custom-docker-builder/Dockerfile]: Will discuss with bamachrn and get back on this.

Comment (mohammedzee1000) [inline images/builder/docker/custom-docker-builder/Dockerfile]: @stevekuznetsov @tdawson 
We are updating the build process to accommodate the tagging issue in FROM stanza[1].
Once this is in, i will update this PR to remove the tags from Dockerfiles

[1] https://github.com/CentOS/container-pipeline-service/pull/330

**Comment by mohammedzee1000** [comment]
@stevekuznetsov @tdawson reverted the dockerfile version changes..

File: images/base/Dockerfile [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-# common to all downstream images should be here. Depends on Centos 7.2+.
+# common to all downstream images should be here. Depends on Centos 7.2+
 #
 # The standard name for this image is openshift/origin-base
 #
-FROM centos:centos7
+FROM registry.centos.org/centos/centos:7
 
-RUN INSTALL_PKGS="which git tar wget hostname sysvinit-tools util-linux bsdtar epel-release \
+RUN INSTALL_PKGS="which git tar wget hostname sysvinit-tools util-linux bsdtar centos-release-openshift-origin15 \
```
Comment (stevekuznetsov) [inline images/base/Dockerfile]: You found that removing `epel-release` didn't break any image builds?

Comment (mohammedzee1000) [reply images/base/Dockerfile]: @stevekuznetsov didn't look like it i was able to successfully do `oc cluster up` with locally built images.

**Comment by mohammedzee1000** [comment]
@stevekuznetsov done

File: images/base/Dockerfile [hunk 4]
```diff
@@ -1,12 +1,12 @@
 #
 # This is the base image from which all OpenShift Origin images inherit. Only packages
-# common to all downstream images should be here. Depends on Centos 7.2+.
+# common to all downstream images should be here. Depends on Centos 7.2+
 #
 # The standard name for this image is openshift/origin-base
 #
-FROM centos:centos7
+FROM registry.centos.org/centos/centos:7
```
Comment (smarterclayton) [inline images/base/Dockerfile]: Does this have better uptime and bandwidth than Dockerhub?

Comment (mohammedzee1000) [reply images/base/Dockerfile]: The registry is meant to always be up, and be the official source of the centos images.
@bamachrn if you want to add ^

Comment (bamachrn) [reply images/base/Dockerfile]: yes, also we are working for making this as a default registry for centos docker.

Comment (mohammedzee1000) [reply images/base/Dockerfile]: @smarterclayton ^

Comment (smarterclayton) [reply images/base/Dockerfile]: When you say default registry... what do you mean?

Comment (mohammedzee1000) [reply images/base/Dockerfile]: We mean that it will be added by default, to `/etc/sysconfig/docker` on future centos releases.
@bamachrn ^

Comment (mohammedzee1000) [reply images/base/Dockerfile]: @smarterclayton ^

Comment (mohammedzee1000) [reply images/base/Dockerfile]: @smarterclayton The registry is meant to always be up, even if the Container Pipeline Service goes down. There are backup registries that get synced regularly and a geo located mirroring that is being worked on as well @bamachrn

Comment (mohammedzee1000) [reply images/base/Dockerfile]: @kbsingh ^

**Pull request closed without merging**